inc/ui-controller.php

```diff
@@ -90,6 +90,15 @@ function process_csv_upload() {
 function store_csv_upload( $move_new_file, $file ) {
 	$page = ! empty( $_POST['slug'] ) ? sanitize_text_field( wp_unslash( $_POST['slug'] ) ) : ''; // wpcs: csrf ok.
 
+	/**
+	 * This handles line endings that don't get parsed correctly. This is
+	 * mostly an issue with older mac versions where the EOL character is \r.
+	 * We replace it with \r\n so that fgetcsv parses the file correctly.
+	 */
+	$content = file_get_contents( $file['tmp_name'] );
+	$content = str_replace( "\r", "\r\n", $content );
```

Comment:
This is going to convert `\r\n` line endings to `\r\r\n`

Reply:
This might be slow, and definitely test it out, but you could potentially...

```php
preg_replace( "/\r$/", "\r\n", $content )
```

_but_, I would first attempt to _detect_ the line endings before replacing them. you might have to do some googlin' to find the best way to do that